pennylane_lightning_gpu/src/bindings/Bindings.cpp

```diff
@@ -85,6 +88,38 @@ void StateVectorCudaManaged_class_bindings(py::module &m) {
             return new StateVectorCudaManaged<PrecisionT>(
                 data_ptr, static_cast<std::size_t>(arr.size()));
         }))
+        .def(
+            "setZeroState",
+            [](StateVectorCudaManaged<PrecisionT> &sv, const size_t index,
+               const bool use_async) {
+                std::complex<PrecisionT> value(0, 0);
+                sv.setState(value, index, use_async);
+            },
+            "Create basis state on GPU.")
+        .def(
+            "setBasisState",
+            [](StateVectorCudaManaged<PrecisionT> &sv, const size_t index,
+               const bool use_async) {
+                std::complex<PrecisionT> value(1, 0);
+                sv.setState(value, index, use_async);
+            },
+            "Create basis state on GPU.")
+        .def(
+            "setStateVector",
+            [](StateVectorCudaManaged<PrecisionT> &sv,
+               const np_arr_sparse_ind &indices, const np_arr_c &state,
+               const bool use_async) {
+                using index_type = typename std::conditional<
+                    std::is_same<ParamT, float>::value, int32_t, int64_t>::type;
+
+                sv.template setStates<index_type>(
+                    static_cast<index_type>(indices.request().size),
+                    static_cast<std::complex<PrecisionT> *>(
+                        state.request().ptr),
+                    static_cast<index_type *>(indices.request().ptr),
+                    use_async);
+            },
+            "Set State Vector on GPU")
```

Comment:
Can this be a little more descriptive? Is this overwriting the value of the GPU statevector from a host numpy array?

Reply:
done